far/vmenu.cpp

```diff
@@ -2996,3 +2982,64 @@ size_t VMenu::Text(wchar_t const Char) const
 {
 	return ::Text(Char, m_Where.width() - (WhereX() - m_Where.left));
 }
+
+#ifdef ENABLE_TESTS
+
+#include "testing.hpp"
+
+TEST_CASE("find.nearest.selectable.item")
+{
+	using namespace std::ranges;
+	using namespace std::views;
+
+	std::array<int, 10> arr{};
+
+	const auto Pred{ [](const int b) { return b != 0; } };
+
+	const auto TestAllPositions{
+		[&](const int Found)
+		{
+			for (const auto Pos : iota(0, static_cast<int>(arr.size())))
+			{
+				REQUIRE(find_nearest(arr, Pos, Pred, false, false) == Found);
+				REQUIRE(find_nearest(arr, Pos, Pred, false, true) == Found);
+				REQUIRE(find_nearest(arr, Pos, Pred, true, false) == Found);
+				REQUIRE(find_nearest(arr, Pos, Pred, true, true) == Found);
+			}
+		} };
+
+	TestAllPositions(-1);
+
+	for (const auto Found : iota(0, static_cast<int>(arr.size())))
+	{
+		fill(arr, int{});
+		arr[Found] = true;
+		TestAllPositions(Found);
+	}
+
+	fill(arr, int{});
+	arr[3] = arr[7] = true;
+
+	static constexpr struct { int Pos; bool GoBackward; bool DoWrap; int Expected; } TestDataPoints[] =
```

Comment:
We don't enforce nonsense like "no more than 80 columns", but it doesn't mean that we should put everything in one line :)
Could you format it like a struct please?